services/galley/src/Galley/API/LegalHold.hs

```diff
@@ -222,11 +221,22 @@ disableForUser
 disableForUser (zusr ::: tid ::: uid ::: req ::: _) = do
     membs <- Data.teamMembers tid
     void $ permissionCheck zusr ChangeLegalHoldUserSettings membs
-    DisableLegalHoldForUserRequest mPassword <- fromJsonBody req
-    ensureReAuthorised zusr mPassword
-
-    Client.removeLegalHoldClientFromUser uid
-    LHService.removeLegalHold tid uid
-    LegalHoldData.setUserLegalHoldStatus tid uid UserLegalHoldDisabled
-    -- send event at this point
-    pure empty
+    if userLHNotDisabled membs
+        then disableLH >> pure empty
+        else pure noContent
+  where
+    -- If not enabled nor pending, then it's disabled
+    userLHNotDisabled mems = do
+        let target = findTeamMember uid mems
+        case fmap (view legalHoldStatus) target of
+            Just UserLegalHoldEnabled -> True
+            Just UserLegalHoldPending -> True
+            _                         -> False
```

Comment:
enumerate all cases so we will catch changes in the data type in the future?  (i think it's only two more, but i haven't checked.)

Reply:
👍